tests/test_invenio_access_permissions.py

```diff
@@ -82,6 +82,41 @@ def test_invenio_access_permission(app):
         assert permission.allows(fake_identity)
 
 
+def test_invenio_access_permission_denied(app):
+    """Permission is always denied when excluded."""
+    fake_identity = FakeIdentity()
+
+    InvenioAccess(app)
+    with app.test_request_context():
+        db.session.begin(nested=True)
+        # Create two users
+        user_one = User(email='test@inveniosoftware.org')
+        user_two = User(email='test2@inveniosoftware.org')
+
+        # Create permission
+        permission = Permission(ActionNeed('read'))
+        # Exclude user one
+        permission.explicit_excludes.add(UserNeed(user_two.id))
+
+        # Check the correct permissions are granted/excluded
+        assert not permission.allows(fake_identity)
+
+        db.session.add(ActionUsers(action='read', user=user_one))
+        db.session.add(ActionUsers(action='read', user=user_two))
+        db.session.commit()
+
+        assert not permission.allows(fake_identity)
+
+        # Allows user one
+        fake_identity.provides.add(UserNeed(user_one.id))
+        assert permission.allows(fake_identity)
+
+        # Denies user two
+        fake_identity.provides.remove(UserNeed(user_one.id))
+        fake_identity.provides.add(UserNeed(user_two.id))
+        assert permission.allows(fake_identity)
```

Comment:
``not``

Reply:
Fixed, ``user.id`` was accessed before commited so it wasnt there.